development/ts-migration-dashboard/app/components/App.tsx

```diff
@@ -73,30 +75,86 @@ export default function App() {
           },
           {},
         );
+  }, [boxRectsByModuleId]);
   const [activeBoxRectId, setActiveBoxRectId] = useState<string | null>(null);
   const activeBoxRect =
     boxRectsByModuleId === null || activeBoxRectId === null
       ? null
       : boxRectsByModuleId[activeBoxRectId];
+  const [searchQuery, setSearchQuery] = useState('');
+  const modulesMatchedBySearch = useMemo(() => {
+    if (searchQuery.trim().length === 0) {
+      return new Set();
+    }
+
+    const searchQueryTokens = searchQuery.trim().split(' ');
+    const modules = partitions.map((partition) => partition.children).flat();
+    return new Set([
+      ...modules.filter((module) => {
+        const moduleIdTokens = module.id.split('/');
+        return searchQueryTokens.some((searchQueryToken) => {
+          return moduleIdTokens.some((moduleIdToken) =>
+            moduleIdToken.startsWith(searchQueryToken),
```

Comment:
- Searching for `"foo"` matches `foo.js` but also `foobar.js`, `foo/bar.js`, etc.
- Searching for `"foo bar"` matches `foo.js` but also `bar.js`, `barbaz.js`, `bar/baz.js`, etc.